packages/sprotty-protocol/src/actions.ts

```diff
@@ -0,0 +1,500 @@
+ */
+export interface LayoutAction {
+    kind: typeof LayoutAction.KIND
+    layoutType?: string
+    elementIds?: string[]
+}
+export namespace LayoutAction {
+    export const KIND = 'layout';
+
+    export function create(options: { layoutType?: string, elementIds?: string[] } = {}): LayoutAction {
+        return {
+            kind: KIND,
+            layoutType: options.layoutType,
+            elementIds: options.elementIds
+        };
+    }
+}
+
+/**
+ * Triggered when the user requests the viewer to center on the current model. The resulting
+ * CenterCommand changes the scroll setting of the viewport accordingly.
+ * It also resets the zoom to its default if retainZoom is false.
+ * This action can also be sent from the model source to the client in order to perform such a
+ * viewport change programmatically.
+ */
+export interface CenterAction extends Action {
+    kind: typeof CenterAction.KIND
+    elementIds: string[]
+    animate: boolean
+    retainZoom: boolean
+}
+export namespace CenterAction {
+    export const KIND = 'center';
+
+    export function create(elementIds: string[], options: { animate?: boolean, retainZoom?: boolean } = {}): CenterAction {
+        return {
+            kind: KIND,
+            elementIds,
+            animate: options.animate ?? true,
+            retainZoom: options.retainZoom ?? false
+        };
+    }
+}
+
+/**
+ * Triggered when the user requests the viewer to fit its content to the available drawing area.
+ * The resulting FitToScreenCommand changes the zoom and scroll settings of the viewport so the model
+ * can be shown completely. This action can also be sent from the model source to the client in order
+ * to perform such a viewport change programmatically.
+ */
+export interface FitToScreenAction extends Action {
+    kind: typeof FitToScreenAction.KIND;
+    elementIds: string[]
+    padding?: number
+    maxZoom?: number
+    animate: boolean
+}
+export namespace FitToScreenAction {
+    export const KIND = 'fit';
+
+    export function create(elementIds: string[], options: { padding?: number, maxZoom?: number, animate?: boolean }): FitToScreenAction {
+        return {
+            kind: KIND,
+            elementIds,
+            padding: options.padding,
+            maxZoom: options.maxZoom,
+            animate: options.animate ?? true
+        };
+    }
+}
+
+/**
+ * Directly set the diagram viewport to the given scroll and zoom values.
+ * The ID of the viewport element to manipulate must be given with the action
+ * (usually it is the root element's ID).
+ */
+export interface SetViewportAction extends Action {
+    kind: typeof SetViewportAction.KIND;
+    elementId: string
+    newViewport: Viewport
+    animate: boolean
+}
+export namespace SetViewportAction {
+    export const KIND = 'viewport';
+
+    export function create(newViewport: Viewport, options: { elementId: string, animate?: boolean }): SetViewportAction {
```

Comment:
I think, either  `newViewport` should be moved into options here or `element` should be moved out of the options object and options should be made optional (default to empty). But the current split into 1 required parameter and 1 required option object doesn't make sense to me